### PR TITLE
Fix custom gas sheet

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-2273 RNBW-2274 RNBW-2271 RNBW-2219

## What changed (plus any additional context for devs)

I fixed the settings of the custom gas sheet on Android. This was a bit of guessing, but after this, I can no longer repro the issue

## PoW (screenshots / screen recordings)

Before:
+ $63 20 ~45 sec
OR:
Max base fee

After: 
Current base fee

## Dev checklist for QA: what to test

Open the gas sheet in DEV / PROD mode, try to dot his quickly / slowly, etc. Test on Android only. Should be no/op on iOS.

### CODE REVIEW

**Comment by skylarbarrera** [comment]
still running into the issue sporadically but now the broken state seems worse (completely black screen), don't think we should have to mess with the margins or content height to fix this.

https://user-images.githubusercontent.com/29204161/150882683-79bacab1-70fb-42f7-b232-c926a19dfa0f.mov

**Review by terrysahaidak — APPROVED** [review]
After testing it on my phone i wasn't able to reproduce it anymore. There are some really minor things on that flow but I think it's not related to this specific PR so i think it at least unblocks us.

File: src/components/gas/GasSpeedButton.js [hunk 1]
```diff
@@ -230,9 +227,21 @@ const GasSpeedButton = ({
     asset,
     shouldOpenCustomGasSheet.focusTo,
     speeds,
-    openCustomOptions,
   ]);
 
+  const openCustomOptions = useCallback(
+    focusTo => {
+      if (ios) {
+        setShouldOpenCustomGasSheet({ focusTo, shouldOpen: true });
```
Comment (estebanmino) [inline src/components/gas/GasSpeedButton.js:235]: just talked with @osdnk this was a nasty hack to make some sheet animations work, makes sense to remove it if it fixes android

**Review by estebanmino — APPROVED** [review]

**Comment by ibrahimtaveras00** [comment]
Still seeing the black screen issue in swaps mode when changing from any speed to custom as well when doing the same in view details sheet; seen here = http://recordit.co/6zbYhLgQQ0

Same thing happens in send flow the first time, but when attempting to change it again it works; seen here = http://recordit.co/JBaDFNfgYq

Also seems to happen during WC flow, and also, when it does end up working, the keyboard is not automatically expanded even though the cursor is in the max base fee input field; seen here = http://recordit.co/6FbC14owxd

**Comment by osdnk** [comment]
This is introducing a small regression on android : https://streamable.com/hmpwjx

The switch changed to "custom" with a small delay. We need to fix this later post-release

**Pull request merged**